person.rb

```diff
@@ -1,21 +1,58 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
+
+class Person < Nameable
   attr_reader :id
   attr_accessor :name, :age
 
-  def initialize(age, parent_permission: true, name: 'unknown')
+  def initialize(age, name = 'unknown', parent_permission: true)
+    super()
     @id = Random.rand(1..1000)
     @name = name
-    @age = age
     @parent_permission = parent_permission
+    @age = age
+    @nameable = Nameable.new
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
   private
 
   def of_age?
-    @age >= 18
+    @age > 18
+  end
+end
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0, 10]
   end
```

Comment:
- [ ] [optional] @Kwaku28 it would have been nice if each class was in its own file, this makes it easier to read the code and upgrade the functions in the future.

Reply:
Updated the file as suggested. Thanks for your input